### PR TITLE
HPCC-19129 Document ESP & Roxie HTTP persistent connections

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>
@afishbeck Please review (I emailed a PDF, too) 

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml [hunk 1]
```diff
@@ -2500,13 +2640,35 @@ sudo -u hpcc cp /etc/HPCCSystems/source/NewEnvironment.xml /etc/HPCCSystems/envi
...
+          <para>Persistent connections can cause backwards compatibility
+          issues with non-standard HTTP clients. You can disable this feature
+          by setting <emphasis>maxHttpConnectionRequests</emphasis> to either
+          0 or 1.</para>
+
+          <para>In addition, persistent connections can affect Roxie load
+          balancing. If this is a concern, you can adjust
+          <emphasis>maxHttpConnectionRequests</emphasis> and
+          <emphasis>maxHttpKeepAliveWait</emphasis> values to optimize how
+          long all the transactions from a particular persistent connection go
```
Comment (afishbeck) [inline docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml]: Makes it sound like we're able to affect the load balancer.  Rather something like:  "to optimize how many persistent connections a particular node will support, and how long they will last."

**Comment by afishbeck** [comment]
There are a lot of whitespace changes that make it hard to find what is actually changing.  But what I found looks ok, with a few comments.
@mayx can you also review?

File: docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml [hunk 2]
```diff
@@ -1552,14 +1610,23 @@ sudo -u hpcc cp /etc/HPCCSystems/source/NewEnvironment.xml /etc/HPCCSystems/envi
...
-          xmlns:xi="http://www.w3.org/2001/XInclude" /></para>
+          <para>
+            <xi:include href="../XMLGeneration/xml/esp.xsd.mod.xml"
+                        xpointer="xpointer(//*[@id='ESP.t6'])"
+                        xmlns:xi="http://www.w3.org/2001/XInclude" />
+          </para>
+
+          <!--***Move to XSD for 2.0***-->
+
+          <para>Persistent HTTP connections can cause backwards compatibility
```
Comment (mayx) [inline docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml]: I don't see any description of the Persistent Connections feature in. general, or explanation of the parameters maxPersistentRequests and maxPersistentIdleTime, are they in a different file that is not included in this PR?

**Review by mayx — COMMENTED** [review]
@JamesDeFabia one comment.

**Comment by richardkchapman** [comment]
@JamesDeFabia

**Comment by mayx** [comment]
PDF doc looks good. So the PR is fine.

**Pull request merged**